doc/developer/repoproviders.rst

```diff
@@ -0,0 +1,60 @@
+.. _providers-section:
+
+====================
+Repository Providers
+====================
+
+Repository Providers (or RepoProviders) are
+locations where git repositories are stored (e.g.,
+GitHub). BinderHub supports a number of providers out of the
+box, and can be extended to support new providers. For a complete
+listing of the provider classes, see :ref:`api-repoproviders`.
+
+Supported repoproviders
+=======================
+
+Currently supported providers, their prefixes and specs are:
+
++------------+--------------------+-------------------------------------------------------------+----------------------------+
+| Provider   | provider_prefix    | spec                                                        | notes                      |
++============+====================+=============================================================+============================+
+| GitHub     | ``gh``             | ``<user>/<repo>/<commit-sha-or-tag-or-branch>``             |                            |
++------------+--------------------+-------------------------------------------------------------+----------------------------+
+| GitLab     | ``gl``             | ``<user>/<repo>/<commit-sha-or-tag-or-branch>``             |                            |
++------------+--------------------+-------------------------------------------------------------+----------------------------+
+| Gist       | ``gist``           | ``<github-username>/<gist-id><commit-sha-or-tag>``          |                            |
++------------+--------------------+-------------------------------------------------------------+----------------------------+
+| Zenodo     | ``zenodo``         | ``<zenodo-DOI>``                                            |                            |
++------------+--------------------+-------------------------------------------------------------+----------------------------+
+| Figshare   | ``figshare``       | ``<figshare-DOI>``                                          |                            |
++------------+--------------------+-------------------------------------------------------------+----------------------------+
+| Git        | ``git``            | ``<url-escaped-url>/<commit-sha>``                          | arbitrary HTTP git repos   |
++------------+--------------------+-------------------------------------------------------------+----------------------------+
+
+Adding a new repository provider
+================================
+
+It is possible to add new repository providers to BinderHub, allowing
+a BinderHub deployment to fetch git repositories from new locations
+on the web. Doing so involves defining your own RepoProvider sub-class
+and modifying a set of methods/attributes to interface with the online
+provider to which you are providing access.
+
+In order to extend the supported repository providers,
```

Comment:
Should we add a step that it is highly likely that your first step would be to add a new content provider to repo2docker?